mobile/src/profile/profile.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { View, LayoutChangeEvent, Text, Image, ImageStyle, StyleSheet } from 'react-native';
+
+import styles from '../layout/styles';
+import { layoutStyles, contentStyles, tileStyles, contactStyles } from './styles';
+import { Chevron } from './chevron';
+import { Avatar } from '../people/avatar';
+import { TopNavBar } from '../topNavBar/top-nav-bar';
+import { AppState } from '../reducers/app.reducer';
+import { UserInfoState } from '../reducers/user/user-info.reducer';
+import { Department } from '../reducers/organization/department.model';
+
+import { StyledText } from '../override/styled-text';
+import { Employee } from '../reducers/organization/employee.model';
+
+interface ProfileProps {
+    employee: Employee;
+    department: Department;
+}
+
+export class Profile extends Component<ProfileProps> {
+    public render() {
+        const employee = this.props.employee;
+        const department = this.props.department;
+
+        if (!employee || !department) {
+            return null;
+        }
+
+        const tilesData: { label: string, icon: any, style: ImageStyle }[] = [
+            {
+                label: employee.birthDate.format('MMMM D'),
+                icon: require('../../src/profile/icons/birthDate.png'),
+                style: StyleSheet.flatten([tileStyles.icon, tileStyles.iconBirthDay])
+            },
+            {
+                label: employee.hireDate.format('YYYY-D-MM'),
+                icon: require('../../src/profile/icons/hireDate.png'),
+                style: StyleSheet.flatten([tileStyles.icon, tileStyles.iconHireDate])
+            },
+            {
+                label: `Room ${employee.roomNumber}`,
+                icon: require('../../src/profile/icons/room.png'),
+                style: StyleSheet.flatten([tileStyles.icon, tileStyles.iconRoom])
+            },
+            {
+                label: 'Organization',
+                icon: require('../../src/profile/icons/organization.png'),
+                style: StyleSheet.flatten([tileStyles.icon, tileStyles.iconOrganization])
+            }
+        ];
+
+        const contactsData: { icon: any, text: string, title: string }[] = [
+            {
+                icon: require('../../src/profile/icons/phone.png'),
+                text: employee ? employee.mobilePhone : '',
+                title: 'Mobile Phone:'
+            },
+            {
+                icon: require('../../src/profile/icons/email.png'),
+                text: employee ? employee.email : '',
+                title: 'Email:'
+            }
+        ];
+
+        const tiles = tilesData.map((tile) => (
```

Comment:
what about moving this into a separate function? 'getTiles(employee)'. I do not fancy 100-lines long functions